src/Robot/SubtaskLibrary.hs

```diff
@@ -47,19 +49,21 @@ subtaskSearch index subtask@(Subtask subtaskType exprID pattern) =
 
 -- Given a destroy task which specifies an expression,
 -- Attempt to achieve the subtask using results from the index
-attemptDestroySubtask :: Index -> Subtask -> AutData -> Move
-attemptDestroySubtask index task@(Subtask subtaskType exprID pattern) autData tab = do
+attemptDestroySubtask :: Index -> Subtask -> Move
+attemptDestroySubtask index task@(Subtask subtaskType exprID pattern) tab = do
+    autData <- getAutData
     guard $ subtaskTypeToSubtaskClass subtaskType == Destroy
     let exprType = subtaskTypeToExprType subtaskType
-    eid <- exprID
+    eid <- liftMaybe exprID
     address <- case exprType of
         H -> getHypFromID eid autData
         T -> getTargFromID eid autData
     let equivalences = subtaskSearch index task
     let invoker = case exprType of
             H -> libEquivHyp
             T -> libEquivTarg
-    listToMaybe $ mapMaybe (\(EquivalenceValue result pair) ->
+    guard $ not $ null equivalences
```

Comment:
This guard wasn't here before - what's it needed for?

Reply:
The next line uses foldl1 which throws an error if equivalences are null